src/screens/Ticketing/Purchase/Overview/index.tsx

```diff
@@ -82,19 +82,33 @@ const PurchaseOverview: React.FC<OverviewProps> = ({
   const {tariffZones, userProfiles} = useFirestoreConfiguration();
 
   const {preassignedFareproducts} = useFirestoreConfiguration();
+  const productType =
+    params.preassignedFareProduct?.type ?? params.selectableProductType;
 
   const selectableProducts = preassignedFareproducts
     .filter(productIsSellableInApp)
-    .filter((product) => product.type === params.selectableProductType);
+    .filter((product) => product.type === productType);
 
   const [preassignedFareProduct, setPreassignedFareProduct] = useState(
-    selectableProducts[0],
+    params.preassignedFareProduct ?? selectableProducts[0],
   );
 
   const defaultUserProfilesWithCount = useDefaultUserProfilesWithCount(
     userProfiles,
     preassignedFareProduct,
   );
+
+  if (params.userProfilesWithCount) {
+    defaultUserProfilesWithCount.forEach((u) => (u.count = 0));
+    params.userProfilesWithCount?.forEach((u: UserProfileWithCount) => {
+      defaultUserProfilesWithCount
+        .filter(
+          (upc: UserProfileWithCount) => upc.userTypeString == u.userTypeString,
+        )
+        .forEach((v: UserProfileWithCount) => (v.count = u.count));
+    });
+  }
+
```

Comment:
I think we can try to simplify this a little bit, also the logic in `useDefaultUserProfilesWithCount`. The logic should be something like this:
- Get the user profiles from Firestore
- Filter by the limitations on the preassigned fare product
- If userProfilesWithCount in screen parameter: Apply them
- If no userProfilesWithCount in screen parameter: Apply default
- Use this as the initial state for travellerSelection

Reply:
I improved this

Reply:
Slightly better, although still kinda complex, but haven't seen into if it actually possible to do in a less complex manner. Approving now!

Reply:
I agree it is complex, I did the easiest possible with the current code solution. I have also changed the naming of some vars for a better understanding of what is going on.